test/geometry/test_conversions.py

```diff
@@ -1631,3 +1634,131 @@ def test_everything(self, device, dtype):
+        q = Quaternion.random(batch_size=1).to(device, torch.float64)
+        assert gradcheck(euler_from_quaternion, (q.w, q.x, q.y, q.z), raise_exception=True)
+
+    def test_module(self, device, dtype):
+        pass
+
+    def test_jit(self, device, dtype):
+        q = Quaternion.random(batch_size=1)
+        q = q.to(device, dtype)
+        op = euler_from_quaternion
+        op_jit = torch.jit.script(op)
+        assert_close(op(q.w, q.x, q.y, q.z), op_jit(q.w, q.x, q.y, q.z))
+
+    def test_forth_and_back(self, device, dtype):
+        q = Quaternion.random(batch_size=2)
+        q = q.to(device, dtype)
+        roll, pitch, yaw = euler_from_quaternion(q.w, q.x, q.y, q.z)
+        qw, qx, qy, qz = quaternion_from_euler(roll, pitch, yaw)
+        # TODO: check hwo to prevent getting inverted angles sometimes
+        assert_close(q.w.abs(), qw.abs())
+        assert_close(q.x.abs(), qx.abs())
+        assert_close(q.y.abs(), qy.abs())
+        assert_close(q.z.abs(), qz.abs())
+
+
+class TestQuaternionFromEuler(BaseTester):
+    def test_smoke(self, device, dtype):
+        roll, pitch, yaw = torch.rand(3, device=device, dtype=dtype)
+        qw, qx, qy, qz = quaternion_from_euler(roll, pitch, yaw)
+        assert qw.shape == qx.shape
+        assert qx.shape == qy.shape
+        assert qy.shape == qz.shape
+
+    @pytest.mark.parametrize("batch_size", ((1, 3, 4)))
+    def test_cardinality(self, device, dtype, batch_size):
+        roll, pitch, yaw = torch.rand(3, batch_size, device=device, dtype=dtype)
+        qw, qx, qy, qz = quaternion_from_euler(roll, pitch, yaw)
+        assert qw.shape[0] == batch_size
+        assert qx.shape[0] == batch_size
+        assert qy.shape[0] == batch_size
+        assert qz.shape[0] == batch_size
+
+    def test_exception(self, device, dtype):
+        _, pitch, yaw = torch.rand(3, 2, device=device, dtype=dtype)
+        with pytest.raises(Exception):
+            quaternion_from_euler(torch.rand(1), pitch, yaw)
+
+    def test_gradcheck(self, device):
+        roll, pitch, yaw = torch.rand(3, 2, device=device, dtype=torch.float64, requires_grad=True)
+        assert gradcheck(quaternion_from_euler, (roll, pitch, yaw), raise_exception=True)
+
+    def test_module(self, device, dtype):
+        pass
+
+    @pytest.mark.skip()
+    def test_jit(self, device, dtype):
+        roll, pitch, yaw = torch.rand(3, 2, device=device, dtype=dtype)
+        op = quaternion_from_euler
+        op_jit = torch.jit.script(op)
+        assert_close(op(roll, pitch, yaw), op_jit(roll, pitch, yaw))
+
+    def test_forth_and_back(self, device, dtype):
+        roll, pitch, yaw = torch.rand(3, 2, device=device, dtype=dtype)
+        qw, qx, qy, qz = quaternion_from_euler(roll, pitch, yaw)
+        roll_new, pitch_new, yaw_new = euler_from_quaternion(qw, qx, qy, qz)
+        assert_close(roll, roll_new)
+        assert_close(pitch, pitch_new)
+        assert_close(yaw, yaw_new)
+
+    def test_values(self, device, dtype):
```

Comment:
I'm getting some failed runs here when testing on cuda.

Reply:
now is fixed